app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -155,5 +156,13 @@ def get_progress_message(action)
       end
       "#{action == 'done' ? 'Published' : action.titleize}: #{action_time} By: #{action_username}"
     end
+
+    def validate_proposal
```

Comment:
As long as the migration is run and the model validation runs, this before action should not be necessary, right?

Reply:
This protects against other sources of bad data as well; humans mucking about in the DB, rails calls that don't respect the validation rules (e.g. update_all).